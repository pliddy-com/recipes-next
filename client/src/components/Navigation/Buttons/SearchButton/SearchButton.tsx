import NavIconButton from '../NavIconButton/NavIconButton';
import SearchIcon from '@mui/icons-material/Search';

interface SearchButtonProps {
  hideLabel?: boolean;
}

const SearchButton = ({ hideLabel }: SearchButtonProps) => {
  const ariaLabel = 'search';
  const className = 'menu';
  const label = 'Search';
  const href = '/search';

  return (
    <NavIconButton
      ariaLabel={ariaLabel}
      className={className}
      hideLabel={hideLabel}
      icon={<SearchIcon />}
      label={label}
      href={href}
    />
  );
};

export default SearchButton;
