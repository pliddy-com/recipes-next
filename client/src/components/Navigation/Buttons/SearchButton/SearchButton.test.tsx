import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

import SearchButton from './SearchButton';

describe('LogoButton', () => {
  const expectedHref = '/search';

  it('renders a logo button with a label', () => {
    const expectedLabel = 'Search';

    const { getByRole } = render(<SearchButton />);

    // assert that the component has been rendered
    const component = getByRole('button', { name: 'search' });

    // assert that the component has correct label
    expect(component.textContent).toContain(expectedLabel);

    // assert that the component has correct href
    expect(component).toHaveAttribute('href', expectedHref);

    // assert that the component matches the existing snapshot
    expect(component).toMatchSnapshot();
  });

  describe('when the hideLabel flag is true', () => {
    it('renders an icon button', () => {
      const { getByRole } = render(<SearchButton hideLabel={true} />);

      // assert that component has been rendered
      const component = getByRole('button', { name: 'search' });

      // assert that the component has correct href
      expect(component).toHaveAttribute('href', expectedHref);

      // assert that the component matches the existing snapshot
      expect(component).toMatchSnapshot();
    });
  });
});
