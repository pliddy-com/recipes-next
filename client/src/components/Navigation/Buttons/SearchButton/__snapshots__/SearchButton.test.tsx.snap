// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoButton renders a logo button with a label 1`] = `
<a
  aria-label="search"
  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButton-colorPrimary menu-button css-kpm0tx-MuiButtonBase-root-MuiButton-root"
  href="/search"
  role="button"
  tabindex="0"
>
  <span
    class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="SearchIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
      />
    </svg>
  </span>
  Search
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</a>
`;

exports[`LogoButton when the hideLabel flag is true renders an icon button 1`] = `
<a
  aria-label="search"
  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium menu-button css-1d6w9lk-MuiButtonBase-root-MuiIconButton-root"
  href="/search"
  role="button"
  tabindex="0"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="SearchIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
    />
  </svg>
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</a>
`;
